src/workspace-names.cpp

```diff
@@ -284,25 +285,22 @@ class wayfire_workspace_names_output : public wf::per_output_plugin_instance_t
         auto wsn     = workspaces[x][y]->workspace;
         int ws_num   = x + y * wsize.width + 1;
 
+        // Get intended name of the workspace
+        std::string tmp_name = output->to_string() + "_workspace_" + std::to_string(ws_num);
+
         if (show_option_names)
         {
-            wsn->name = output->to_string() + "_workspace_" +
-                std::to_string(ws_num);
+            wsn->name = tmp_name;
         } else
         {
             bool option_found = false;
-            for (auto option : section->get_registered_options())
+            for (const auto& [wsid, wsname] : workspace_names.value())
             {
-                int ws;
-                if (sscanf(option->get_name().c_str(),
-                    (output->to_string() + "_workspace_%d").c_str(), &ws) == 1)
+                if (wsid == tmp_name)
```

Comment:
I haven't tested this, but it seems like this doesn't do what the original code intended.